skillserver/skillserver.go

```diff
@@ -291,14 +291,26 @@ func IsValidAlexaRequest(w http.ResponseWriter, r *http.Request) bool {
 }
 
 func readCert(certURL string) ([]byte, error) {
-	cert, err := http.Get(certURL)
+	certPool, err := x509.SystemCertPool()
+	insecureSkipVerify := false
+	if err != nil || certPool == nil {
+		log.Println("Can't open system cert pools, doing insecure skip verify instead")
+		insecureSkipVerify = true
+	}
+
+	tr := &http.Transport{
+		TLSClientConfig: &tls.Config{RootCAs: certPool, InsecureSkipVerify: insecureSkipVerify},
+	}
+	hc := &http.Client{Timeout: 2 * time.Second, Transport: tr}
+
+	cert, err := hc.Get(certURL)
 	if err != nil {
-		return nil, errors.New("Could not download Amazon cert file.")
+		return nil, errors.New("Could not download Amazon cert file. Reason: " + err.Error())
```

Comment:
A valid follow up question: why would you ever want to skip ssl verification here?

Reply:
Thanks, I will add that.
Insecure skip verify is definitely dangerous, but I was thinking if somebody got a warning, they could atleast go about testing the the thing they made. Definitely not recommended for production cases. There are sometimes cases with docker containers built from scratch that root ca certs aren't present. So I thought if you're only testing, it would be okay to just log the error but not stop the whole thing altogether. I like the idea of keeping a flag and saying explicitly, if you wanted to do insecure skip verify, like curl does it